src/test/client/nonblocking.cc

```diff
@@ -712,3 +714,76 @@ TEST_F(TestClient, LlreadvContiguousLlwritevNonContiguous) {
   client->ll_release(fh);
   ASSERT_EQ(0, client->ll_unlink(root, filename, myperm));        
 }
+
+TEST_F(TestClient, LlreadvLlwritevLargeBuffers) {
+  /* Test that async I/O code paths handle large buffers (total len >= 4GiB)*/
+  int mypid = getpid();
+  char filename[256];
+
+  client->unmount();
+  TearDown();
+  SetUp();
+
+  sprintf(filename, "test_llreadvllwritevlargebuffers%u", mypid);
+
+  Inode *root, *file;
+  root = client->get_root();
+  ASSERT_NE(root, (Inode *)NULL);
+
+  Fh *fh;
+  struct ceph_statx stx;
+
+  ASSERT_EQ(0, client->ll_createx(root, filename, 0666,
+                                  O_RDWR | O_CREAT | O_TRUNC,
+                                          &file, &fh, &stx, 0, 0, myperm));
+
+  struct statvfs stbuf;
+  int64_t rc;
+  rc = client->ll_statfs(root, &stbuf, myperm);
+  ASSERT_EQ(rc, 0);
+  int64_t fs_available_space = stbuf.f_bfree * stbuf.f_bsize;
+  ASSERT_GT(fs_available_space, 0);
+
+  const int64_t BUFSIZE = 1024 * 1024 * 1024;
+  int64_t bytes_written = 0, bytes_read = 0;
+
+  std::unique_ptr<C_SaferCond> writefinish = nullptr;
+  std::unique_ptr<C_SaferCond> readfinish = nullptr;
+  writefinish.reset(new C_SaferCond("test-nonblocking-writefinish-large-buffers"));
+  readfinish.reset(new C_SaferCond("test-nonblocking-readfinish-large-buffers"));
+
+  auto out_buf_0 = std::make_unique<char[]>(BUFSIZE * 3);
```

Comment:
Just a minor comment, instead of using`BUFSIZE * 3`, one could use use `BUFSIZE`, and then if you made` BUFSIZE = INT_MAX + 1` it would be clearer to read.

Reply:
Done PTAL https://github.com/ceph/ceph/compare/e491ff3bd1ff7726af5480739a865970ef5c7261..f590eea329e4e589452a061dbd7b9b8de715c1b5